addons/mail/static/src/new/messaging_menu/messaging_menu.js

```diff
@@ -72,10 +72,10 @@ export class MessagingMenu extends Component {
 
         const tab = this.state.tab;
         if (tab === "all") {
-            return previews;
+            return previews.reverse();
         }
         const target = this.tabToThreadType(tab);
-        return previews.filter((preview) => target.includes(preview.type));
+        return previews.filter((preview) => target.includes(preview.type)).reverse();
```

Comment:
Order was wrong in test so you just reversed the order, but this is still wrong: ordering of threads should be based on date of last message, not the order of insertion.

Reply:
Done ! 
Btw message.dateTime is already a dateTime object :)